spec/validate-body-decorator.spec.ts

```diff
@@ -0,0 +1,156 @@
+		return jsonWithMessage(`Hello ${body.name}`);
+	}
+
+	@Post('/body-with-prop')
+	sayHelloToHim2(@Body('name') body: DTO) {
+		return jsonWithMessage(`Hello ${body}`);
+	}
+
+	@Post('/multiples-bodys-parameters')
+	sayHelloToThem(@Body('name') body1: DTO, @Body() body2: DTO) {
+		return jsonWithMessage(
+			`Hello ${body1} and ${body2.name}, wait... it's the same people`,
+		);
+	}
+}
+
+@Module({
+	controllers: [AppController],
+})
+class App {}
+
+Deno.test('Controller works', async () => {
+	const app = new DanetApplication();
+	await app.init(App);
+	const port = (await app.listen(0)).port;
+
+	const res = await fetch(`http://localhost:${port}/test`, {
+		method: 'GET',
+	});
+	const json = await res.json();
+	assertEquals(json, jsonWithMessage('Hello'));
+	await app.close();
+});
+
+Deno.test('Test', async (t) => {
+	const app = new DanetApplication();
+	await app.init(App);
+	const port = (await app.listen(0)).port;
+	let res, json;
+
+	// ------------------------------------------------------------------
+	// STEP 1 => Return normal if body is equals DTO and 400 if it is not
+	// ------------------------------------------------------------------
+
+	// RIGHT
+	res = await fetchWithBody(`http://localhost:${port}/test`, {
+		name: 'James',
+		age: 25, // A string as number
+	});
+	assertEquals(res.status, 200);
+	assertEquals(await res.json(), jsonWithMessage('Hello James'));
+
+	// WRONG
+	res = await fetchWithBody(`http://localhost:${port}/test`, {
+		name: 'James',
+		age: 'Potter', // A string as number
+	});
+	assertEquals(res.status, 400);
+
+	json = await res.json();
+	assertExists(json);
+	assertExists(json.reasons);
+
+	// -----------------------------------------------------------
+	// STEP 2 => Same as last one, but passing a prop in @Body(..)
+	// -----------------------------------------------------------
+
+	// WRONG
+	res = await fetchWithBody(`http://localhost:${port}/test/body-with-prop`, {
+		name: 'James',
+		age: 'Potter', // A string as number
+	});
+	assertEquals(res.status, 400);
+	json = await res.json();
+	assertExists(json);
+	assertExists(json.reasons);
+
+	// -----------------------------------------------------------------------
+	// STEP 3 => Multiples @Body, one extract a prop, and other the whole body
+	// -----------------------------------------------------------------------
+
+	// WRONG
+	res = await fetchWithBody(
+		`http://localhost:${port}/test/multiples-bodys-parameters`,
+		{
+			name: 'James',
+			age: 'Potter', // A string as number
+		},
+	);
+	assertEquals(res.status, 400);
+	json = await res.json();
+	assertExists(json);
+	assertExists(json.reasons);
+
+	// RIGHT
+	res = await fetchWithBody(
+		`http://localhost:${port}/test/multiples-bodys-parameters`,
+		{
+			name: 'James',
+			age: 25,
+		},
+	);
+	assertEquals(res.status, 200);
+	json = await res.json();
+	assertEquals(
+		json,
+		jsonWithMessage('Hello James and James, wait... it\'s the same people'),
+	);
+
+	await app.close();
+});
```

Comment:
Some tests are redundant with `method-param-decorator.test.ts` @Body tests, I think you may want to modify the existing tests and add this failing test.

```
// WRONG
	res = await fetchWithBody(`http://localhost:${port}/test`, {
		name: 'James',
		age: 'Potter', // A string as number
	});
	assertEquals(res.status, 400);
```

Reply:
Also, it's better IMO if you create multiple Deno.test with meaningful description instead of writing the description in comments